local-volume/provisioner/pkg/deleter/blockcleaner_test.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deleter
+
+import (
+	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/cache"
+	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/common"
+	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/util"
+	"io/ioutil"
+	"k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/record"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
```

Comment:
Can you also add a test for duplicate operations?

Reply:
@msau42 Can you clarify more on what duplicate operation means here? Do you mean specify some command twice in a single exec call and see if it actually ran twice? or do you mean run the same command in 2 iterations? There is a test for latter. I wasn't quite sure what the former scenario is attempting to test.

Reply:
Since we perform the operation asynchronously, the main control loop will invoke the cleanup on this PV every few sounds, especially if the cleanup takes a long time.  So a test to make sure that we drop the subsequent requests while the 1st is still ongoing would be good.